sdk/samples/iot/paho_iot_hub_pnp_component_sample.c

```diff
@@ -867,10 +867,18 @@ static void process_twin_message(
           exit(rc);
         }
 
+        rc = az_iot_pnp_client_twin_property_builder_begin_component(
+            &pnp_client, &jw, component_name);
+        if (az_result_failed(rc))
```

Comment:
I noticed a lot of error logging on failure and exiting.  This would be a great place I think to use `IOT_SAMPLE_EXIT_IF_AZ_FAILED`.  It should save you 5 lines each time I think.

Example:
```
rc = az_iot_pnp_client_twin_property_builder_begin_component(
            &pnp_client, &jw, component_name);
IOT_SAMPLE_EXIT_IF_AZ_FAILED(rc, "Could not begin the property component");
```

OR

```
IOT_SAMPLE_EXIT_IF_AZ_FAILED(az_iot_pnp_client_twin_property_builder_begin_component(
            &pnp_client, &jw, component_name), "Could not begin the property component");
```

Reply:
yup good call updating

Reply:
I fixed these. I will issue another PR which searches for any `if (az_result_failed` in the pnp samples and update.

Reply:
They don't necessarily have to be everywhere, like in the standard sample functions that are also checking for the return type of the MQTT calls, i think this format is fine because it parallels the MQTT checks.